### PR TITLE
Initial load of Ibrahim's ASP.NET Webforms post

### PR DESCRIPTION
**Pull request by leebrandt**

### CODE REVIEW

File: _source/_posts/2018-08-29-secure-webforms-with-openidconnect-okta.md [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+We talk a lot about ASP.NET Core on this blog, but the .NET ecosystem is much much more than just Core. Microsoft is still updating and supporting .NET Framework, and will continue to do so for the foreseeable future. The same is the case with Web Forms, which even got some new features recently. In this post, we'll take a look at how to easily secure existing ASP.NET Web Forms with via an external provider — Okta.
+
+To follow along, you'll need .NET Framework 4.7.1 and Visual Studio 2017.
+
+
+## Get Started with ASP.NET and Web Forms
+
+First, you need to create a new Web Forms application using one of the built-in templates that ship with Visual Studio. Choose the Web Forms template with Individual User Accounts authentication.  
+
+In Visual Studio, select File -> New Project -> Name the project AspNetWebFormsOkta
```
Comment (aaronpk) [inline _source/_posts/2018-08-29-secure-webforms-with-openidconnect-okta.md]: Bold these

**File**

**New Project**

File: _source/_posts/2018-08-29-secure-webforms-with-openidconnect-okta.md [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+Uninstall-Package Microsoft.ApplicationInsights.PerfCounterCollector
+Uninstall-Package Microsoft.ApplicationInsights.DependencyCollector
+Uninstall-Package Microsoft.ApplicationInsights.Agent.Intercept
+Uninstall-Package Microsoft.ApplicationInsights
+Uninstall-Package Microsoft.AspNet.TelemetryCorrelation
+```
+
+You can also delete `ApplicationInsights.config` file.
+
+You'll want to specify the port that your app will be running on, so you can use it later during your Okta configuration. To do so, right click on the project in the solution explorer and click **properties**. In the main properties window, choose **Web** from the left-hand menu and set the **Project Url** property to `http://localhost:8080`.
```
Comment (aaronpk) [inline _source/_posts/2018-08-29-secure-webforms-with-openidconnect-okta.md]: verify that the menu item says **Url** and not **URL**

File: _source/_posts/2018-08-29-secure-webforms-with-openidconnect-okta.md [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+
+On the settings page, enter the following values:
+* Name: OktaWebForms
+* Grant type allowed: Implicit (Hybrid)
+
+You can leave the other values unchanged, and click **Done**.
+
+Now that your application has been created, copy down the Client ID and Client secret values on the following page, you'll need them soon (of course, yours will be different).
+
+{% img blog/aspnet-webforms/okta-client-credentials.png alt:"Okta client credentials" width:"800" %}{: .center-image }
```
Comment (aaronpk) [inline _source/_posts/2018-08-29-secure-webforms-with-openidconnect-okta.md]: I hope you've deleted this application already!

File: _source/_posts/2018-08-29-secure-webforms-with-openidconnect-okta.md [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+And much more! Check out our [product documentation](https://developer.okta.com/documentation/)
+
+## Secure Your ASP.NET Application
+You should store your Okta details inside your `Web.config`, so you can use those later on in our code. To do this, make use of the `appSettings` section of the configuration.
+
+You want to add a connection string to `Web.config` file. Make sure to add it inside of `<configuration>`, before the `<system.web></system.web>` section:
+
+```cs
+<appSettings>
+  <add key="okta:ClientId" value="{clientId}" />
```
Comment (aaronpk) [inline _source/_posts/2018-08-29-secure-webforms-with-openidconnect-okta.md]: IIRC this should be `{yourClientId}` and `{yourClientSecret}` to be replaced properly

File: _source/_posts/2018-08-29-secure-webforms-with-openidconnect-okta.md [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+```
+
+Next, you will make use of your Okta credentials and Okta's `UseOpenIdConnectAuthentication` middleware to connect to an externally hosted authorization server. You definitely don't want to build and maintain your own authorization server, so Okta will handle token generation for you.
+
+
+### Display a Login Button
+
+Next you'll add a `Login` button to your existing navigation bar. Inside of the `Site.Master` file, after the existing `<ul>` tag, add the following:
+
+```cs
```
Comment (aaronpk) [inline _source/_posts/2018-08-29-secure-webforms-with-openidconnect-okta.md]: shouldn't this be `html`?

File: _source/_posts/2018-08-29-secure-webforms-with-openidconnect-okta.md [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+mainContent.Controls.Add(label);
+```
+
+Now you can start the application! After the page loads click on Login button, you will be redirected to the Okta login page. After entering your login details and signing in, you will be redirected back to the application. 
+
+Now go to the About page to see if you can see the protected data you added:
+
+{% img blog/aspnet-webforms/running-application.png alt:"Application Running" width:"800" %}{: .center-image }
+
+The complete and working sample that backs this tutorial is available on GitHub as [AspNetWebFormsOkta](https://github.com/Ibro/AspNetWebFormsOkta)
```
Comment (aaronpk) [inline _source/_posts/2018-08-29-secure-webforms-with-openidconnect-okta.md]: add period

File: _source/_posts/2018-08-29-secure-webforms-with-openidconnect-okta.md (hunk 2 above)
Comment (leebrandt) [reply _source/_posts/2018-08-29-secure-webforms-with-openidconnect-okta.md]: It does

File: _source/_posts/2018-08-29-secure-webforms-with-openidconnect-okta.md (hunk 3 above)
Comment (leebrandt) [reply _source/_posts/2018-08-29-secure-webforms-with-openidconnect-okta.md]: :)

**Pull request merged**